MeshLib/MeshGenerators/LayeredMeshGenerator.cpp

```diff
@@ -56,7 +61,20 @@ MeshLib::Mesh* LayeredMeshGenerator::getMesh(std::string const& mesh_name) const
     if (_nodes.empty() || _elements.empty())
         return nullptr;
 
-    MeshLib::Mesh* result (new MeshLib::Mesh(mesh_name, _nodes, _elements));
+	MeshLib::Properties properties;
+	if (!_materials.empty())
+	{
+		if (_materials.size() == _elements.size())
+		{
+			boost::optional<MeshLib::PropertyVector<int> &> materials = 
+				properties.createNewPropertyVector<int>("MaterialIDs", MeshLib::MeshItemType::Cell);
+			materials->resize(_materials.size());
+			std::copy(_materials.cbegin(), _materials.cend(), materials->begin());
+		}
+		else
+			ERR ("Skipping MaterialID information, number of entries does not match element number");
```

Comment:
Would suggest to abort on any errors, but if not an error, just emit a warning...